lightning/src/chain/channelmonitor.rs

```diff
@@ -3681,6 +3694,39 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 		Ok(())
 	}
 
+	fn promote_funding(&mut self, new_funding_txid: Txid) -> Result<(), ()> {
+		let is_pending_splice = self.pending_funding.iter().any(|funding| funding.is_splice());
+
+		let new_funding = self
+			.pending_funding
+			.iter_mut()
+			.find(|funding| funding.funding_txid() == new_funding_txid);
+		if new_funding.is_none() {
+			return Err(());
+		}
+		let mut new_funding = new_funding.unwrap();
+
+		// `first_confirmed_funding_txo` is set to the first outpoint for the channel upon init.
+		// If an RBF happens and it confirms, this will no longer be accurate, so update it now
+		// if we know the RBF doesn't belong to a splice.
+		if !is_pending_splice && self.first_confirmed_funding_txo == self.funding.funding_outpoint()
+		{
+			self.first_confirmed_funding_txo = new_funding.funding_outpoint();
+		}
+
+		mem::swap(&mut self.funding, &mut new_funding);
+		self.onchain_tx_handler.update_after_renegotiated_funding_locked(
+			self.funding.current_holder_commitment_tx.clone(),
+			self.funding.prev_holder_commitment_tx.clone(),
+		);
+
+		for funding in self.pending_funding.drain(..) {
```

Comment:
Might be worth noting that the `swap` above allows us to clean up both the previous funding and any pending RBFs.